core/store/orm/orm_test.go

```diff
@@ -1330,6 +1330,67 @@ func TestORM_SyncDbKeyStoreToDisk(t *testing.T) {
 	assert.Equal(t, key.JSON.String(), content)
 }
 
+const linkEthTxWithTaskRunQuery = `
+INSERT INTO eth_task_run_txes (task_run_id, eth_tx_id) VALUES ($1, $2)
+`
+
+func TestORM_RemoveUnstartedTransaction(t *testing.T) {
+	storeInstance, cleanup := cltest.NewStore(t)
+	defer cleanup()
+	ormInstance := storeInstance.ORM
+
+	jobSpec := cltest.NewJobWithRunLogInitiator()
+	require.NoError(t, storeInstance.CreateJob(&jobSpec))
+
+	for _, status := range []models.RunStatus{
+		models.RunStatusInProgress,
+		models.RunStatusUnstarted,
+	} {
+		jobRun := cltest.NewJobRun(jobSpec)
+		jobRun.Status = status
+		jobRun.TaskRuns = []models.TaskRun{
+			{
+				ID:         models.NewID(),
+				Status:     models.RunStatusUnstarted,
+				TaskSpecID: jobSpec.Tasks[0].ID,
+			},
+		}
+		runRequest := models.NewRunRequest(models.JSON{})
+		require.NoError(t, storeInstance.DB.Create(&runRequest).Error)
+		jobRun.RunRequest = *runRequest
+		require.NoError(t, storeInstance.CreateJobRun(&jobRun))
+
+		key := cltest.MustInsertRandomKey(t, storeInstance)
+		ethTx := cltest.MustInsertUnconfirmedEthTxWithBroadcastAttempt(t, storeInstance, 1, key.Address.Address())
+		require.NoError(t, storeInstance.DB.Exec(linkEthTxWithTaskRunQuery, jobRun.TaskRuns[0].ID.UUID(), ethTx.ID).Error)
+	}
+
+	assert.NoError(t, ormInstance.RemoveUnstartedTransactions())
+
+	jobRuns, err := ormInstance.JobRunsFor(jobSpec.ID, 10)
+	assert.NoError(t, err)
+	assert.Len(t, jobRuns, 1, "expected only one JobRun to be left in the db")
+	assert.Equal(t, jobRuns[0].Status, models.RunStatusInProgress)
+
+	taskRuns := []models.TaskRun{}
+	assert.NoError(t, storeInstance.DB.Find(&taskRuns).Error)
+	assert.Len(t, taskRuns, 1, "expected only one TaskRun to be left in the db")
+
+	runRequests := []models.RunRequest{}
+	assert.NoError(t, storeInstance.DB.Find(&runRequests).Error)
+	assert.Len(t, runRequests, 1, "expected only RunRequest to be left in the db")
+
+	// TODO fix these tests
```

Comment:
I can't seem to link EthTxes with TaskRuns, I tried with a special `linkEthTxWithTaskRunQuery` but I can't figure out what format the TaskRun ID should be - is is UUID?

Reply:
It's a `models.ID`. Very similar to a UUID, but without the dashes so that it can fit into a `bytes32` on Ethereum.